src/mpi/dbcsr_mpiwrap.F

```diff
@@ -2136,17 +2135,19 @@ SUBROUTINE mp_get_processor_name(procname, resultlen)
 
       CHARACTER(LEN=*), INTENT(OUT)                      :: procname
          !! Name of processor, declared as CHARACTER(LEN=mp_max_processor_name)
-      INTEGER, INTENT(OUT)                               :: resultlen
+      INTEGER, OPTIONAL, INTENT(OUT)                     :: resultlen
          !! Length (in characters) of procname (INTEGER)
 
 #if defined(__parallel)
-      INTEGER                                            :: ierr
+      INTEGER                                            :: namelen, ierr
 
-      CALL mpi_get_processor_name(procname, resultlen, ierr)
-      IF (ierr /= 0) THEN
+      CALL mpi_get_processor_name(procname, namelen, ierr)
+      IF (ierr .EQ. 0) THEN
+         IF (PRESENT(resultlen)) resultlen = namelen
+      ELSE
```

Comment:
hmm, we could simply drop the whole `mp_get_processor_name` function, it seems unused?

We should definitely implement the linker script for the public API, build a shared lib using `-Wl,--as-needed` and then check which symbols get dropped in the process. Or maybe easier: collect all exported (`PUBLIC`) symbols and match them against `USE ..., ONLY: `?

Reply:
So, we actually need that function... Currently, we use the `m_hostnm` function directly, for example [here](https://github.com/cp2k/dbcsr/blob/42ad7bc98c88b118d1cbd1a12152b780f809d9b9/src/core/dbcsr_log_handling.F#L560). Ideally, we should replace that call with the more generic MPI call... 
Anyway, I like the idea of the linker to check the unused symbols...

Reply:
I have private/future changes that rely on `mp_get_processor_name`. Beside of Alfio's point (it's in-use), I believe this function can be also of future/further use.